api-remove.go

```
@@ -74,6 +78,110 @@ func (c Client) RemoveObject(bucketName, objectName string) error {
 	return nil
 }
 
+type objectNameError struct {
+	objectName string
+	err        error
+}
+
+// generateRemoveMultiObjects - generate the XML request for remove multi objects request
+func generateRemoveMultiObjectsRequest(objects []string) []byte {
+	rmObjects := []deleteObject{}
+	for _, obj := range objects {
+		rmObjects = append(rmObjects, deleteObject{Key: obj})
+	}
+	xmlBytes, _ := xml.Marshal(deleteMultiObjects{Objects: rmObjects, Quiet: true})
+	return xmlBytes
+}
+
+// processRemoveMultiObjectsResponse - parse the remove multi objects web service
+// and return the success/failure result status for each object
+func processRemoveMultiObjectsResponse(body io.Reader, objects []string, errorCh chan objectNameError) {
+	// Parse multi delete XML response
+	rmResult := &deleteMultiObjectsResult{}
+	respBody, _ := ioutil.ReadAll(body)
```

Let's not ignore the error here..